src/maxminddb.c

```diff
@@ -394,15 +400,22 @@ LOCAL int read_metadata(MMDB_s *mmdb)
         .offset = 0
     };
 
-    mmdb->metadata.node_count =
-        value_for_key_as_uint32(&metadata_start, "node_count");
+    int status =
+        value_for_key_as_uint32(&metadata_start, "node_count",
+                                &mmdb->metadata.node_count);
+    if (MMDB_SUCCESS != status) {
+        return status;
+    }
     if (!mmdb->metadata.node_count) {
         DEBUG_MSG("could not find node_count value in metadata");
         return MMDB_INVALID_METADATA_ERROR;
     }
 
-    mmdb->metadata.record_size =
-        value_for_key_as_uint16(&metadata_start, "record_size");
+    status = value_for_key_as_uint16(&metadata_start, "record_size",
```

Comment:
Here is the [PR that changed `record_size`](https://github.com/maxmind/MaxMind-DB-Writer-perl/pull/1/files#diff-515a62dee46d0cba148ebb060724f9a4L427).